docs_app/content/tutorial/4-what-is-an-operator.md

```diff
@@ -0,0 +1,71 @@
+
+## Containers
+
+The idea of a [container](DICT) is simple. It holds things. Okay, easy right? But an interesting thing about containers is, some types of containers can hold "inner containers" of the same type. Boxes inside of boxes. Arrays inside of arrays. Observables inside of observables.
+
+If you have a container type, that can contains a collection of the same type of containers, that container can be _flattened_. That is to say, that you could systematically look in a container, pull out any containers you find, and dump their contents into a new container of the same type.
+
+Visually, in JavaScript that might look like this (shown with array literals):
+
+```ts
+[[1, 2, 3], [4, 5], [6]].flat() === [1, 2, 3, 4, 5, 6]
+```
+
+In terms of our apples above, that would be a box of boxes of apples, that you could then take out and dump into a new box of just apples.
+
+## Collections And Containers In JavaScript
+
+There are quite a few collections and containers already in existence in JavaScript. It's worth noting a few of them for comparisons here:
+
+### Arrays
+
+The most common collections of things we deal with in JavaScript are [Arrays](MDN_LINK_HERE). Arrays are collections that can be accessed synchronously, and randomly at indices. You can [map](MDN) an array into a new array of the same size. You can create an array of a smaller size (provided your current array isn't length of `0`), so it can be [filtered](MDN) to a new array. And arrays have a definitive "end" to them, which is represented in array literals with `]`, and at runtime by checking to see if your index is outside of the bounds provided by the `length`, therefor arrays can be [concatenated](MDN). Arrays can also be [flattened](MDN), as you can have an array of arrays.
+
+### Promises
+
+Yes, believe it or not, a promise is indeed a "collection". It's a collection of exactly one thing, no more no less. That means they're slightly more limited, but they can be mapped, with [`then`](MDN), concatenated (also `then`), and flattened (which is done automatically with, you guessed it... `then`). However, they can't be placed "inside" of one another. And they can't be "filtered", (there has to be one value or a rejection). So without even getting into their lack of cancellation, it's fair to say they're limited in terms of available operations.
+
+### Iterables
+
+There is a whole bit that we could go into here about how [Observable is the "dual" of Iterable](FIND_LINK), however, for brevity's sake: Iterables are [pull](GL)-based collections/containers of values, in which you can use an [Iterator](MDN) (or iteration via `for..of`) in order to [pull](GL) one value out at a time. Iterables can "contain" other Iterables. They can be any size. And Iterables signal their end by providing a `done` value in an [iteration result](MDN). They can be mapped, filtered, flattened, concatenated, et al. 
```

Comment:
Is this first sentence needed here? It seems like this is referring to something that the average reader might not know ...  like a private joke of sorts. If it is needed, consider moving the sentence down, later in this paragraph, after describing what iterables are.